### DISCUSSION TITLE
Something weird with compression-webpack-plugin

### DISCUSSION THREAD

**Discussion by vralle** [webdiscus/html-bundler-webpack-plugin#133] [Q&A] [answered]
Some thoughts about Web Vitals led me to an experimental Webpack configuration, where [compression-webpack-plugin](https://github.com/webpack-contrib/compression-webpack-plugin) generates compressed versions of static files (html, js, сss, svg, json...). I used the plugin without additional settings. As a result of compilation, pre-compressed versions of static files were added to the result. Tests of the compiled html pages did not reveal any errors. I deployed the result on the server and was surprised to find an error loading js files. The message in the console informed that the js file was not loaded from the link to the source code (!). I was very surprised and the first thing I did was recheck the Webpack configuration. The settings were correct - for production. Then I rechecked the compiled html pages. And I also did not find any errors. The link to the js file fully corresponded to the production configuration. But the error in the browser console pointed to a completely different link used in the source code of the web page. I was already at a complete loss when I unzipped the compressed copy of the page and found this damn link to the source code there.

I spent some more time transforming the link in the source code of the pages, including the inline request, but it did not lead to anything. The links to JS files in compressed archives still point to the source code instead of the compiled files.

**Comment by vralle** [comment]
During the experiments, a scenario for another error was found.  There are several html pages using the same app.js file. But if you change the script tag in one of the templates (it can be an async attribute or an inline request), then the compilation fails. It's not critical in my scenarios, but

**Reply by webdiscus** [reply]
Hello @vralle,

> But if you change the script tag in one of the templates (it can be an async attribute or an inline request), then the compilation fails.

I can't reproduce the issue. Can you please create a small repo and describe the steps how to reproduce it?

**Comment by webdiscus** [comment]
@vralle
 Can you please create a small repo with reproducible  issue by using  compression-webpack-plugin?

**Comment by vralle** [comment]
Hi, @webdiscus and Merry Christmas!

I created an "issues" branch [here](https://github.com/vralle/webpack-html-bundler-starting-template/tree/issues)
If we use `webpack.config.mjs`, the html file in `.gz` archive will not be minified. 
If we use `webpack.edge.mjs`, the link to js file in the html file in `.gz` archive will point to the source.

**Reply by webdiscus** [reply, marked as answer]
Hello @vralle,

Thank you!
Merry Christmas to you too!

Please, follow the issue #134.
